app/services/reachable_users_service.rb

```diff
@@ -110,4 +110,51 @@ def get_users_count
     query = ActiveRecord::Base.send(:sanitize_sql_array, [sql, params])
     ActiveRecord::Base.connection.execute(query).to_a.first
   end
+
+  def self.get_running_campaigns_for_user(user)
+    sql = <<~SQL.tr("\n", " ").squish
+      select
+      c.id as campaign_id,
+      c.available_doses,
+      COALESCE(cm.n_confirmed_matches, 0) as n_confirmed_matches,
+      COALESCE(um.n_user_matches, 0) as n_user_matches,
+      c.min_age,
+      c.max_age,
+      c.status,
+      v.name,
+      c.max_distance_in_meters/1000 as max_distance,
+      (SQRT(((v.lat - (:lat))*110.574)^2 + ((v.lon - (:lon))*111.320*COS((:lat)::float*3.14159/180))^2)) as distance
+      from campaigns c
+      left outer join vaccination_centers v on (v.id = c.vaccination_center_id)
+      LEFT JOIN (
+          SELECT campaign_id, COUNT(*) AS n_confirmed_matches
+          FROM matches
+          WHERE confirmed_at IS NOT NULL
```

Comment:
I would add a filter here, otherwise you'll be running a group by on all matches ever, which is going to be long (group is done before join).

One way to do this would be to join on all currently active campaigns only before grouping, should be fast since at any one time there are not _that_ many campaigns running

eg
```
SELECT campaign_id, COUNT(*) AS n_confirmed_matches
          FROM 
                    matches m
                    campaigns c ON (c.id=m.campaign_id)
          WHERE 
                    m.confirmed_at IS NOT NULL
                    AND c.created_at <= NOW() AND c.ends_at > NOW()
GROUP BY 1
```

or something like that.